### PR TITLE
IPC/etheral recharge with inducers

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
IPCs (and i suppose any future subclass with TRAIT_POWERHUNGRY) can now be recharged with an inducer shot at their face, or by using an inducer in-hand

## Why It's Good For The Game
It seems like an oversight that IPC's can recharge with any electrical recharger besides the inducer, so i went and fixed that up. It also helps medstaff/robotics/engineers for reviving/healing purposes

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl:
add: IPC chassis's (and ethereals) have been modified to interface with heavy-duty inducers
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
I like it

**Review by Penwin0 — COMMENTED** [review]
These changes also apply to ethereals although I'm not certain if they should, update title and CL accordingly please.

File: code/game/objects/items/inducer.dm [hunk 1]
```diff
@@ -104,22 +104,34 @@
...
 	var/obj/O
 	var/coefficient = 1
+	var/obj/item/organ/stomach/battery/battery
 	if(istype(A, /obj/item/gun/energy))
 		to_chat(user,"Error unable to interface with device")
 		return FALSE
 	if(istype(A, /obj))
 		O = A
-	if(C)
+	if(istype(A, /mob/living/carbon))
```
Comment (Penwin0) [inline code/game/objects/items/inducer.dm]:
```suggestion
	if(iscarbon(A))
```

File: code/game/objects/items/inducer.dm [hunk 2]
```diff
@@ -104,22 +104,34 @@
...
 			recharging = FALSE
 			return TRUE
 		user.visible_message("[user] starts recharging [A] with [src].","<span class='notice'>You start recharging [A] with [src].</span>")
-		while(C.charge < C.maxcharge)
+		while((battery?.charge || C.charge) < maxcharge)
 			if(do_after(user, 10, target = user) && cell.charge)
 				done_any = TRUE
-				induce(C, coefficient)
+				if(battery)
+					battery.adjust_charge_scaled(10)  // pure guesstimate of a good recharge amt
```
Comment (Penwin0) [inline code/game/objects/items/inducer.dm]:
```suggestion
					battery.adjust_charge(min(cell.charge,250))
```
This is the most common ammount. Make sure you drain the Inducer cell as well

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**